File: airesources/Scala/hlt/Log.scala
package hlt

import java.io.{FileWriter, IOException}

object Log {
  private var writer: FileWriter = _

  def log(message: String): Unit =
    try {
      writer.write(message)
      writer.write('\n')
      writer.flush()
    } catch {
      case e: IOException =>
        e.printStackTrace()
    }

  private[hlt] def initialize(f: FileWriter): Unit = writer = f
}

File: airesources/Java/hlt/Log.java
package hlt;

import java.io.FileWriter;
import java.io.IOException;

public class Log {

    private final FileWriter file;
    private static Log instance;

    private Log(final FileWriter f) {
        file = f;
    }

    static void initialize(final FileWriter f) {
        instance = new Log(f);
    }

    public static void log(final String message) {
        try {
            instance.file.write(message);
            instance.file.write('\n');
            instance.file.flush();
        }
        catch (IOException e) {
            e.printStackTrace();
        }
    }
}
